src/test/java/org/openmrs/addonindex/domain/AddOnInfoAndVersionsTest.java

```diff
@@ -40,5 +39,23 @@ public void testSerialization() throws Exception {
 		assertThat(parsed.getVersions().size(), is(1));
 		assertThat(parsed.getVersions().get(0).getVersion().toString(), is("1.0"));
 	}
+
+	@Test
+    public void setDetailsBasedOnLatestVersion(){
+        AddOnVersion version = new AddOnVersion();
+        version.setVersion(new Version("2.0"));
+        version.setModuleId("1");
+        version.setModulePackage("org.openmrs.module.openmrs");
+
+        AddOnInfoAndVersions info = new AddOnInfoAndVersions();
+        info.setName("OpenMRS");
+        info.setDescription("Write code, save lives");
+        info.addVersion(version);
```

Comment:
After making the changes I mentioned earlier in the code review, you'll want to have this test put _two_ versions on AddOnInfoAndVersions and ensure it uses the right one.